Add IL2001-IL2060 trimmer warnings (#25842)
dotnet/docs: docs/core/deploying/trim-warnings/il2022.md

---
title: "IL2022: Could not find matching constructor for custom attribute specified in custom attribute annotations file"
description: "Learn about trim warning IL2022: CustomAttributeAnnotationsConstructorNotFound"
ms.date: 08/25/2021
ms.topic: reference
author: mateoatr
ms.author: matorre
f1_keywords:
  - "IL2022"
---
# IL2022: Could not find matching constructor for custom attribute specified in custom attribute annotations file

## Cause

The constructor of a custom attribute specified in a custom attribute annotations file
could not be found.

## Rule description

[Custom attribute annotation files](https://github.com/mono/linker/blob/main/docs/data-formats.md#custom-attributes-annotations-format)
are used to instruct the trimmer to behave as if the specified item has a given

> attribute. Attribute annotations can only be used to add attributes which have effect on

Reviewer: ```suggestion
attribute. Attribute annotations can only be used to add attributes that have an effect on
```

> the trimmer behavior, all other attributes will be ignored. Attributes added via

Reviewer: ```suggestion
the trimmer behavior; all other attributes are ignored. Attributes added via
```
